monitor-function-calls/src/agent.spec.js

```diff
@@ -0,0 +1,621 @@
+      mockTrace[0].action.from = validContractAddress;
+
+      // update mock transaction event with new mock trace and correct contract address
+      mockTxEvent.traces = mockTrace;
+      mockTxEvent.transaction.to = validContractAddress;
+
+      // eliminate any expression from the configuration file
+      initializeData.contracts.forEach((contract) => {
+        const { functionSignatures } = contract;
+        functionSignatures.forEach((functionSignature) => {
+          if (functionSignature.functionName === testConfig.functionInConfig.name) {
+            // these delete statements will still work even if the keys don't exist
+            /* eslint-disable no-param-reassign */
+            delete functionSignature.expression;
+            delete functionSignature.expressionObject;
+            /* eslint-enable no-param-reassign */
+          }
+        });
+      });
+
+      // run the handler
+      const findings = await handleTransaction(mockTxEvent);
+
+      const argumentData = {};
+      argNames.forEach((name, argIndex) => {
+        if (argumentData[name] == null) {
+          argumentData[name] = argValues[argIndex].toString();
+        }
+      });
+
+      // create the expected finding
+      const testFindings = [Finding.fromObject({
+        alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-FUNCTION-CALL`,
+        description: `The ${testConfig.functionInConfig.name} function was invoked in the ${contractName} contract`,
+        name: `${config.protocolName} Function Call`,
+        protocol: config.protocolName,
+        severity: FindingSeverity[testConfig.findingSeverity],
+        type: FindingType[testConfig.findingType],
+        metadata: {
+          contractAddress: validContractAddress,
+          contractName,
+          functionName: testConfig.functionInConfig.name,
+          ...argumentData,
+        },
+      })];
+
+      expect(findings).toStrictEqual(testFindings);
+    });
+
+    it('returns a finding if a target contract invokes a monitored function when the expression condition is met', async () => {
+      let expression;
+      let expressionObject;
+      initializeData.contracts.forEach((contract) => {
+        const { functionSignatures } = contract;
+        functionSignatures.forEach((functionSignature) => {
+          if (functionSignature.functionName === testConfig.functionInConfig.name) {
+            expressionObject = functionSignature.expressionObject;
+            expression = functionSignature.expression;
+          }
+        });
+      });
+      const { variableName, operator, value } = expressionObject;
+
+      // encode function data
+      // valid function name with valid arguments
+      const { argValues, argNames } = createMockArgs(testConfig.functionInConfig);
+
+      // override the argument value that corresponds to the expression condition
+      const argOverride = { name: variableName };
+      if (BigNumber.isBigNumber(value)) {
```

Comment:
Pulling this `if/else if/else` and nested `switch/case` statements out of the test and into a separate helper function would make it easier to read the test